### PR TITLE
Jcf queue dequeue set map / it-academy.by / M-JC1-69-21 / Davydov Bozhen / done tasks "Map", "Set"

### PR DESCRIPTION
**Pull request by davydovbv**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [ ] fixed
- [ ] approved

### CODE REVIEW

**Comment by davydovbv** [comment]
Empty message

пн, 3 мая 2021 г., 16:27 Dmitry Rakovets ***@***.***>:

File: src/main/java/com/rakovets/course/java/core/practice/jcf_map/TextMonitoring.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
+package com.rakovets.course.java.core.practice.jcf_map;
+
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class TextMonitoring {
+    HashMap<String, Integer> counterWords;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/jcf_map/TextMonitoring.java:8]:
```suggestion
    private Map<String, Integer> counterWords;
```

File: src/main/java/com/rakovets/course/java/core/practice/jcf_map/ValueseComporator.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+public class ValueseComporator implements Comparator<Map.Entry<String, Integer>> {
+    @Override
+    public int compare(Map.Entry<String, Integer> o1, Map.Entry<String, Integer> o2) {
+        return o1.getValue().compareTo(o2.getValue());
+    }
+
+
+
+
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/jcf_map/ValueseComporator.java:15]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/main/java/com/rakovets/course/java/core/practice/jcf_set/web_service/Server.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
+package com.rakovets.course.java.core.practice.jcf_set.web_service;
+
+import com.rakovets.course.java.core.practice.jcf_set.web_service.operations.Operation;
+import com.rakovets.course.java.core.practice.jcf_set.web_service.operations.comaparator.OperationsComparator;
+import java.time.LocalTime;
+import java.util.*;
+
+public class Server {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/jcf_set/web_service/Server.java:9]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/main/java/com/rakovets/course/java/core/practice/jcf_set/web_service/UsersDataBase.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    public UsersDataBase(Map<String, String> loginAndPasswords) {
+        this.loginAndPasswords = loginAndPasswords;
+    }
+
+    public void addNewUser(User user) {
+        if (!loginAndPasswords.containsKey(user.getLogin())) {
+            loginAndPasswords.put(user.getLogin(), user.getPassword());
+        }
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/jcf_set/web_service/UsersDataBase.java:20]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/main/java/com/rakovets/course/java/core/practice/jcf_set/web_service/UsersDataBase.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
+package com.rakovets.course.java.core.practice.jcf_set.web_service;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class UsersDataBase {
+    Map<String, String> loginAndPasswords = new HashMap<>();
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/jcf_set/web_service/UsersDataBase.java:7]:
```suggestion
    private Map<String, String> loginAndPasswords = new HashMap<>();
```

File: src/main/java/com/rakovets/course/java/core/practice/jcf_set/web_service/input_output_tools/LoginAndPasswordInputTool.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+   public LoginAndPasswordInputTool(UsersDataBase dataBase) {
+       scanner = new Scanner(System.in);
+       this.dataBase = dataBase;
+   }
+
+    public String chooseLogin()  {
+        System.out.println("Please input Login");
+        return scanner.nextLine();
+    }
+    public String choosePassword() {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/jcf_set/web_service/input_output_tools/LoginAndPasswordInputTool.java:21]: "Violation: There isn't separating empty line
Solution: Add blank line"

File: src/main/java/com/rakovets/course/java/core/practice/jcf_set/web_service/input_output_tools/LoginAndPasswordInputTool.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        System.out.println("Please choose Password");
+        return scanner.nextLine();
+    }
+
+    public String inputLogin() throws LoginDoesNotExistException {
+       System.out.println("Login:");
+       String login = scanner.nextLine();
+       if (!dataBase.getLoginAndPasswords().containsKey(login)){
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/jcf_set/web_service/input_output_tools/LoginAndPasswordInputTool.java:29]: "Violation: There isn't separating white spaces for curly brake
Solution: Add white spaces"

File: src/main/java/com/rakovets/course/java/core/practice/jcf_set/web_service/input_output_tools/LoginAndPasswordInputTool.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+    public String inputNewLogin() throws LoginAlreadyExistsException {
+       System.out.println("New Login");
+       String login =  scanner.nextLine();
+       if (dataBase.getLoginAndPasswords().containsKey(login)){
+           throw new LoginAlreadyExistsException();
+       }
+       return login;
+    }
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/jcf_set/web_service/input_output_tools/LoginAndPasswordInputTool.java:48]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/main/java/com/rakovets/course/java/core/practice/jcf_set/web_service/input_output_tools/MenuButtonInputTool.java [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+import java.util.Scanner;
+
+public class MenuButtonInputTool {
+    static Scanner scanner = new Scanner(System.in);
+
+    public static int button() throws WrongButtonNumberException {
+        System.out.println("Button:");
+        int button = scanner.nextInt();
+        scanner.nextLine();
+        if (button < 0 || button > 5){
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/jcf_set/web_service/input_output_tools/MenuButtonInputTool.java:13]: "Violation: Useless blank lines
Solution: Remove blank lines"

File: src/main/java/com/rakovets/course/java/core/practice/jcf_set/web_service/operations/comaparator/OperationsComparator.java [hunk 10]
```diff
@@ -0,0 +1,12 @@
+package com.rakovets.course.java.core.practice.jcf_set.web_service.operations.comaparator;
+
+import com.rakovets.course.java.core.practice.jcf_set.web_service.operations.Operation;
+import java.util.Comparator;
+
+public class OperationsComparator implements Comparator<Operation> {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/jcf_set/web_service/operations/comaparator/OperationsComparator.java:7]: "Violation: Useless blank lines
Solution: Remove blank lines"

**Review by rakovets — CHANGES_REQUESTED** [review]
You need to fix the following problems.

**Pull request closed without merging**